packages/components/pie-button/src/button.scss

```diff
@@ -2,37 +2,47 @@
 
 .o-btn {
     border-radius: dt.$radius-rounded-e;
-    border: none;
+    border: 1px solid transparent;
     font-family: dt.$font-interactive-m-family;
     font-weight: dt.$font-interactive-m-weight;
+
+    cursor: pointer;
+    user-select: none;
+    outline: none;
+    box-shadow: none;
+
+    &:focus-visible {
+        box-shadow: 0 0 0 2px dt.$color-focus;
+    }
 }
 
 // TODO - Use latest design tokens for font size and line height
-//        once fozzie has been updated to handle them.
+// once fozzie has been updated to handle them.
 
+/* NOTE: button padding-y is calculated by excluding border-width (1px) */
 .o-btn--xsmall {
-    padding: 6px dt.$spacing-b;
+    padding: 5px dt.$spacing-b;
```

Comment:
This doesn't seem ideal. Is there another way around this? Could we use outline instead of border, perhaps?

Reply:
Hey @xander-marjoram, the PR isn't ready yet and I'm still playing around with different approaches :) thanks for the heads up tho.

Reply:
Done ✨

Reply:
so the main issue I was trying to solve is that the `outline` variant doesn't have the same height as other variants. this is because we're adding a border, which by default is placed outside of the element content, resulting in an extra 2px. 
to overcome this issue in Snacks and Fozzie, we subtract 1px from top/bottom padding values, however, I agree with you, it feels a bit hacky. 

I think one way to have the border included in the height is to use the `box-shadow` property to simulate inner borders. [commit](https://github.com/justeattakeaway/pie/pull/208/commits/2eda74f13fc0b625ebc825918639b7cd05c29599)

Alternatively, we can use `outline` but I'm afraid we might introduce layout edge cases as they are drawn outside of the element, and some old browsers don't support rounded outlines. 👀

Reply:
For what it's worth I don't think we're supporting browsers old enough to have issues with `outline` :)